app/src/main/java/ch/epfl/qedit/backend/database/FirebaseDBService.java

```diff
@@ -1,49 +1,180 @@
 package ch.epfl.qedit.backend.database;
 
 import androidx.annotation.NonNull;
-import ch.epfl.qedit.util.BundledData;
+import ch.epfl.qedit.model.Question;
+import ch.epfl.qedit.model.Quiz;
 import ch.epfl.qedit.util.Callback;
 import ch.epfl.qedit.util.Response;
 import com.google.android.gms.tasks.OnCompleteListener;
 import com.google.android.gms.tasks.Task;
 import com.google.firebase.firestore.DocumentSnapshot;
 import com.google.firebase.firestore.FirebaseFirestore;
-import java.util.Objects;
+import com.google.firebase.firestore.QueryDocumentSnapshot;
+import com.google.firebase.firestore.QuerySnapshot;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
 
 public class FirebaseDBService implements DatabaseService {
 
-    private final FirebaseFirestore db;
+    private FirebaseFirestore db;
 
     public FirebaseDBService() {
         // Access a Cloud Firestore instance
         db = FirebaseFirestore.getInstance();
     }
 
+    private Question getQuestionFromDoc(QueryDocumentSnapshot doc) throws Exception {
+
+        String title = doc.get("title", String.class);
```

Comment:
We could maybe refactor each group of two lines into a function. E.g. `String title = getField("title", String.class)`. The exception message could be built from the attribute name, something like: attribute title was not found in the document.

Reply:
let's do this!

Reply:
thanks